### PR TITLE
Adding support for max31865 driver for pt100 thermal sensor.

### PR DESCRIPTION
**Pull request by kornelu123**

### CODE REVIEW

File: max318xx.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  sleep_ms(1);
+
+  return dst[1];
+}
+
+
+#if CONFIG_THERMO == pt100
+  #include "max31865.h"
+#elif COFNIG_THERMO == KType
+  #include "max31856.h"
```
Comment (SecMeant) [inline max318xx.h:63]:
```C
#else
    #error "Invalid CONFIG_THERMO value"
#endif
```
?

File: max318xx.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  spi_write_read_blocking(MAX318xx_SPI_INSTANCE, src, dst, len);
+  gpio_put(FURNACE_SPI_CSN_PIN, 1);
+
+  sleep_ms(1);
+
+  return dst[1];
+}
+
+
+#if CONFIG_THERMO == pt100
```
Comment (SecMeant) [inline max318xx.h:60]: We probably have to also change `Makefile`, no?
[Right now](https://github.com/SecMeant/pico-furnace/blob/ebbb880052bde688c9a7fbb8bdab7f69f13ba604/Makefile#L7) we are setting `CONFIG_THERMO := 1` by default.

**Review by SecMeant — CHANGES_REQUESTED** [review]
Looks great. Have two tiny comments.

File: max318xx.h (hunk 2 above)
Comment (kornelu123) [reply max318xx.h:60]: Yeah, that sound's right, there was also a typo, already fixed. ;]

**Review by SecMeant — APPROVED** [review]
Ship it!

**Pull request merged**